$OpenBSD$

too many arguments to function error with our ports LLVM
override causes an error (non-virtual function)
wrong namespace/class for getNewUninitMemBuffer

Index: Utilities/JIT.cpp
--- Utilities/JIT.cpp.orig
+++ Utilities/JIT.cpp
@@ -109,7 +109,7 @@ extern void jit_finalize()
 
 	for (auto&& t : s_unfire)
 	{
-		mem.deregisterEHFrames(std::get<0>(t), std::get<1>(t), std::get<2>(t));
+		mem.deregisterEHFrames();
 	}
 
 	s_unfire.clear();
@@ -293,7 +293,7 @@ struct MemoryManager : llvm::RTDyldMemoryManager
 		return RTDyldMemoryManager::registerEHFrames(addr, load_addr, size);
 	}
 
-	void deregisterEHFrames(u8* addr, u64 load_addr, std::size_t size) override
+	void deregisterEHFrames(u8* addr, u64 load_addr, std::size_t size)
 	{
 	}
 };
@@ -380,7 +380,7 @@ class ObjectCache final : public llvm::ObjectCache (pu
 	{
 		if (fs::file cached{path, fs::read})
 		{
-			auto buf = llvm::MemoryBuffer::getNewUninitMemBuffer(cached.size());
+			auto buf = llvm::WritableMemoryBuffer::getNewUninitMemBuffer(cached.size());
 			cached.read(const_cast<char*>(buf->getBufferStart()), buf->getBufferSize());
 			return buf;
 		}
